samples/tutorial/consumer/src/main.rs

```diff
@@ -0,0 +1,178 @@
+use samples_protobuf_data_access::tutorial_grpc::v1::{GetRequest, InvokeRequest};
+use serde_derive::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tokio::time::{sleep, Duration};
+use tonic::Status;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ShowNotificationRequestPayload {
+    #[serde(rename = "Notification")]
+    notification: sdv::hmi::show_notification::request::TYPE,
+    #[serde(rename = "$metadata")]
+    metadata: Metadata,
+}
+
+/// Start the Get Signals Repeater.
+/// This will call the get operation on the digital twin provider synchronously to obtain the entity value.
+///
+/// # Arguments
+/// `provider_uri_map` - The provider uri map where the key is the entity id and the value is the provider's uri.
+async fn start_get_signals_repeater(
+    provider_uri_map: HashMap<String, String>,
+) -> Result<(), Status> {
+    debug!("Starting the Consumer's get signals repeater.");
+
+    loop {
+        for (entity, provider_uri) in &provider_uri_map {
+            let response = retry_async_based_on_status(30, Duration::from_secs(1), || {
+                send_get_request(provider_uri, entity)
+            })
+            .await?;
+
+            info!("Get response for entity {entity}: {response}");
+        }
+
+        debug!("Completed sending the get requests.");
+
+        sleep(Duration::from_secs(5)).await;
+    }
+}
+
+/// Start the show-notification repeater.
+///
+/// # Arguments
+/// `provider_uri` - The provider_uri.
+async fn start_show_notification_repeater(provider_uri: String) -> Result<(), Status> {
+    debug!("Starting the consumer's show-notification repeater.");
+
+    let metadata = Metadata { model: sdv::hmi::show_notification::request::ID.to_string() };
+
+    let request_payload: ShowNotificationRequestPayload = ShowNotificationRequestPayload {
+        notification: "Hello world notification.".to_string(),
+        metadata,
+    };
+
+    let request_payload_json = serde_json::to_string(&request_payload).unwrap();
+
+    loop {
+        info!("Sending an invoke request on entity {} with payload '{}' to provider URI {provider_uri}",
+            sdv::hmi::show_notification::ID, &request_payload_json);
+
+        let client_result = DigitalTwinProviderTutorialClient::connect(provider_uri.clone()).await;
+        if client_result.is_err() {
+            warn!("Unable to connect. We will retry in a moment.");
+            sleep(Duration::from_secs(1)).await;
+            continue;
+        }
+        let mut client = client_result.unwrap();
+
+        let request = tonic::Request::new(InvokeRequest {
+            entity_id: sdv::hmi::show_notification::ID.to_string(),
+            payload: request_payload_json.to_string(),
+        });
+
+        let response = client.invoke(request).await?;
+
+        info!("Show notification response: {}", response.into_inner().response);
+        debug!("Completed the invoke request");
+        sleep(Duration::from_secs(15)).await;
```

Comment:
Perhaps add a blank line above both the sleep call and the debug call, as they are associated with separate logical steps.